ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml

```diff
@@ -88,176 +87,192 @@ StatusDialog {
-            Layout.topMargin: 2
-            Layout.preferredHeight: payPanel.height + receivePanel.height + 4
+            Item {
+                Layout.fillWidth: true
+                Layout.margins: 2
+                Layout.preferredHeight: payPanel.height + receivePanel.height + 4
 
-            SwapInputPanel {
-                id: payPanel
-                objectName: "payPanel"
+                SwapInputPanel {
+                    id: payPanel
+                    objectName: "payPanel"
 
-                anchors {
-                    left: parent.left
-                    right: parent.right
-                    top: parent.top
-                }
+                    anchors {
+                        left: parent.left
+                        right: parent.right
+                        top: parent.top
+                    }
 
-                currencyStore: root.swapAdaptor.currencyStore
-                flatNetworksModel: root.swapAdaptor.swapStore.flatNetworks
-                processedAssetsModel: root.swapAdaptor.walletAssetsStore.groupedAccountAssetsModel
+                    currencyStore: root.swapAdaptor.currencyStore
+                    flatNetworksModel: root.swapAdaptor.swapStore.flatNetworks
+                    processedAssetsModel: root.swapAdaptor.walletAssetsStore.groupedAccountAssetsModel
+                    plainTokensBySymbolModel: root.plainTokensBySymbolModel
 
-                tokenKey: root.swapInputParamsForm.fromTokensKey
-                tokenAmount: root.swapInputParamsForm.fromTokenAmount
+                    tokenKey: root.swapInputParamsForm.fromTokensKey
+                    tokenAmount: root.swapInputParamsForm.fromTokenAmount
 
-                selectedNetworkChainId: root.swapInputParamsForm.selectedNetworkChainId
-                selectedAccountAddress: root.swapInputParamsForm.selectedAccountAddress
-                nonInteractiveTokensKey: receivePanel.selectedHoldingId
+                    selectedNetworkChainId: root.swapInputParamsForm.selectedNetworkChainId
+                    selectedAccountAddress: root.swapInputParamsForm.selectedAccountAddress
+                    nonInteractiveTokensKey: receivePanel.selectedHoldingId
 
-                swapSide: SwapInputPanel.SwapSide.Pay
-                swapExchangeButtonWidth: swapExchangeButton.width
+                    swapSide: SwapInputPanel.SwapSide.Pay
+                    swapExchangeButtonWidth: swapExchangeButton.width
 
-                onSelectedHoldingIdChanged: root.swapInputParamsForm.fromTokensKey = selectedHoldingId
-                onValueChanged: {
-                    if(root.swapInputParamsForm.fromTokensKey === selectedHoldingId) {
-                        root.swapInputParamsForm.fromTokenAmount = !tokenAmount && value === 0 ? "" : value.toLocaleString(locale, 'f', -128)
+                    onSelectedHoldingIdChanged: root.swapInputParamsForm.fromTokensKey = selectedHoldingId
+                    onRawValueChanged: {
```

Comment:
Fixed 2 issues here:
- wrong decimal separator
- precision loss by working on `double value` instead of the full BigInt